app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,71 @@
+class LikesController < ApplicationController
+  before_action :authenticate_member!, except: :index
+
+  respond_to :html, :json
+
+  def create
+    @like = Like.new
+    @like.member = current_member
+    @like.likeable = find_likeable
+
+    respond_to do |format|
+      if @like.save
+        format.html { redirect_to @like.likeable }
+        format.json do
+          render(
+            json: {
+              id: @like.likeable.id,
+              liked_by_member: true,
+              description: ActionController::Base.helpers.pluralize(@like.likeable.likes.count, "like"),
+              url: like_path(@like, format: :json)
+            },
+            status: 201
+          )
+        end
+      else
+        format.html do
+          flash[:error] = 'Unable to like'
+          redirect_to @like.likeable
+        end
+      end
+    end
+  end
+
+  def destroy
```

Comment:
same again. Can be smaller? Some of the json rendering can go elsewhere and therefore be standardised.